go/types/list.go

```diff
@@ -170,33 +173,35 @@ func (l List) IterAll(f listIterAllFunc) {
 	}
 }
 
-func (l List) copyReadAhead(out []Value, startIdx uint64) (numBytes uint64) {
-	llen := l.Len()
-	d.PanicIfFalse(startIdx < llen)
+func copyReadAhead(col Collection, blockLength uint64, startIdx uint64) (out []Value, numBytes uint64) {
```

Comment:
let's make this `iterRange(f func(V Value), start, end)` or `getRange(start, end)`. you choose.